tests/python/pants_test/backend/python/tasks/test_python_tool.py

```diff
@@ -0,0 +1,91 @@
+# coding=utf-8
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import glob
+import os
+import re
+
+from pants.backend.python.subsystems.python_tool_base import PythonToolBase
+from pants.backend.python.tasks.python_tool_prep_base import PythonToolInstance, PythonToolPrepBase
+from pants.task.task import Task
+from pants.util.collections import assert_single_element
+from pants.util.contextutil import environment_as, temporary_dir
+from pants_test.backend.python.tasks.python_task_test_base import PythonTaskTestBase
+
+
+class Tool(PythonToolBase):
+  options_scope = 'test-tool'
+  default_requirements = [
+    'pex==1.5.3',
+  ]
+  default_entry_point = 'pex.bin.pex:main'
+
+
+class ToolInstance(PythonToolInstance):
+  pass
+
+
+class ToolPrep(PythonToolPrepBase):
+  options_scope = 'tool-prep-task'
+  tool_subsystem_cls = Tool
+  tool_instance_cls = ToolInstance
+
+
+class ToolTask(Task):
+  options_scope = 'tool-task'
+
+  @classmethod
+  def prepare(cls, options, round_manager):
+    super(ToolTask, cls).prepare(options, round_manager)
+    round_manager.require_data(ToolPrep.tool_instance_cls)
+
+  def execute(self):
+    tool_for_pex = self.context.products.get_data(ToolPrep.tool_instance_cls)
+    stdout, stderr, exit_code, _ = tool_for_pex.output(['--version'])
+    assert '' == stderr
+    assert re.match(r'.*\.pex 1.5.3', stdout)
+    assert 0 == exit_code
+
+
+class PythonToolPrepTest(PythonTaskTestBase):
+
+  @classmethod
+  def task_type(cls):
+    return ToolTask
+
+  def _assert_tool_execution_for_python_version(self, use_py3=True):
+    scope_string = '3' if use_py3 else '2'
+    constraint_string = 'CPython>=3' if use_py3 else 'CPython<3'
+    tool_prep_type = self.synthesize_task_subtype(ToolPrep, 'tp_scope_py{}'.format(scope_string))
+    context = self.context(for_task_types=[tool_prep_type], for_subsystems=[Tool], options={
+      'test-tool': {
+        'interpreter_constraints': [constraint_string],
+      },
+    })
+    # XDG_CACHE_HOME overrides the location of the cache dir.
```

Comment:
You plumb options for test-tool just above. Consider doing the same for `'': {'pants_bootstrapdir': tmp_dir}`. XDG_CACHE_HOME is a another - less fragile - implementation detail; this time of:
https://github.com/pantsbuild/pants/blob/e382541d70c5b57e682dce998d361c96aed4d8e9/src/python/pants/option/global_options.py#L150-L151

Reply:
I was...very tired when I first wrote the test case. Will also make this change.